### PR TITLE
feat/login with backend

### PR DESCRIPTION
**Pull request by zo02**
Hook up frontend login with backend authentication endpoints

### CODE REVIEW

File: gui/Test/src/main/java/sample/test/controllers/LoginViewController.java [hunk 1]
```diff
@@ -58,26 +63,45 @@ public void cancelButtonOnAction(ActionEvent event) {
...
+
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(loginUrl))
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(jsonInputString))
+                    .build();
+
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+            if (response.statusCode() == 200) {
```
Comment (blackspyek) [inline gui/Test/src/main/java/sample/test/controllers/LoginViewController.java]: Extract conditional to method

File: gui/Test/src/main/java/sample/test/controllers/LoginViewController.java [hunk 2]
```diff
@@ -33,6 +36,8 @@ public class LoginViewController implements Initializable {
     @FXML
     private PasswordField enterPasswordField;
 
+    private String jwtToken;
```
Comment (blackspyek) [inline gui/Test/src/main/java/sample/test/controllers/LoginViewController.java:40]: I noticed that the current implementation of the JWT token in the LoginViewController has a limitation: the token does not persist across view changes or reloads. When the view is reloaded or the user navigates to another screen, the token will be lost, which can lead to authentication issues.

To address this, I recommend implementing a singleton pattern or a dedicated service to manage the JWT token. This way, we can ensure that the token remains accessible throughout the application lifecycle.

**Review by blackspyek — COMMENTED** [review]
You need to fix it!

File: gui/Test/src/main/java/sample/test/controllers/LoginViewController.java (hunk 2 above)
Comment (zo02) [reply gui/Test/src/main/java/sample/test/controllers/LoginViewController.java:40]: Got it. Thanks for the feedback, I will fix it right away.

**Comment by blackspyek** [comment]
LGTM

**Pull request merged**